esp/esdllib/EsdlAccessMapGenerator.hpp

```diff
@@ -0,0 +1,153 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2013 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifndef _EsdlAccessMapGenerator_HPP_
+#define _EsdlAccessMapGenerator_HPP_
+
+#include "AccessMapGenerator.hpp"
+#include "esdl_def.hpp"
+#include "espcontext.hpp"
+#include "jlog.hpp"
+#include "jscm.hpp"
+#include "seclib.hpp"
+
+using EsdlAccessMapGenerator = TAccessMapGenerator<SecAccessFlags>;
+using EsdlAccessMapScopeMapper = EsdlAccessMapGenerator::ScopeMapper;
+
+struct EsdlAccessMapLevelMapper : public EsdlAccessMapGenerator::LevelMapper
+{
+    SecAccessFlags levelUnavailable() const override { return SecAccess_Unavailable; }
+    SecAccessFlags levelNone() const override { return SecAccess_None; }
+    SecAccessFlags levelDeferred() const override { return SecAccess_None; }
+    SecAccessFlags levelAccess() const override { return SecAccess_Access; }
+    SecAccessFlags levelRead() const override { return SecAccess_Read; }
+    SecAccessFlags levelWrite() const override { return SecAccess_Write; }
+    SecAccessFlags levelFull() const override { return SecAccess_Full; }
+    SecAccessFlags levelUnknown() const override { return SecAccess_Unknown; }
+
+    bool isEqual(SecAccessFlags lhs, SecAccessFlags rhs) const override
+    {
+        return lhs == rhs;
+    }
+
+    const char* toString(SecAccessFlags level) const override
```

Comment:
Could you instead use seclib.hpp `inline const char* getSecAccessFlagName(SecAccessFlags flag)...`